### PR TITLE
docs(config) document optimization.innerGraph config option

### LINKED ISSUE #3273: Document webpack change: refactor Parser to allow variable tagging, add Deep Scope Analysis (inner graph) for unused exports
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/9642

---

use variable tagging in harmony plugin (instead of magic renaming)
Parser no longer calls hooks.expression for x in `x = ...`, hooks.pattern is called instead
remove rootName argument from hooks.expressionMemberChain and hooks.callMemberChain

export declarations are executed in block pre walking instead of pre walking

This PR integrates the work of @vincentdchan of GSoC 2018, which allows webpack follow module internal function and pure expressions during unused export analysis.

fixes #6264

**What kind of change does this PR introduce?**
refactoring, bugfixes, feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
new option `optimization.innerGraph` to enabled/disable inner graph analysis. Default: on in production mode, otherwise off

### PR DESCRIPTION
**Pull request by EugeneHlushko**
As per title.
Closes #3273

### CODE REVIEW

**Comment by montogeek** [comment]
Can we add something else to say why is this important or how it is useful?

File: src/content/configuration/optimization.md [hunk 1]
```diff
@@ -519,3 +519,22 @@ module.exports = {
   }
 };
 ```
+
+## `optimization.innerGraph`
+
+`bool`
+
+`optimization.innerGraph` tells webpack whether to conduct inner graph analysis.
```
Comment (EugeneHlushko) [inline src/content/configuration/optimization.md]:
```suggestion
`optimization.innerGraph` tells webpack whether to conduct inner graph analysis for unused exports.
```

**Comment by EugeneHlushko** [comment]
This is related: https://github.com/webpack/webpack/issues/6264
Although i am not sure what to add as we'd want people to just have it "on" for production builds when tree shaking going on iirc

@sokra can you add anything ?

**Comment by montogeek** [comment]
I guess we can add something else later, this is fine.

**Comment by EugeneHlushko** [comment]
All right, i've mentioned that its for unused exports so it makes more sense than before. Merging

**Pull request merged**